pkg/controller/gcp/database/instance.go

```diff
@@ -0,0 +1,236 @@
+/*
+Copyright 2018 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package database
+
+import (
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2/google"
+	sqladmin "google.golang.org/api/sqladmin/v1beta4"
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/crossplaneio/crossplane/pkg/apis/gcp/database/v1alpha1"
+	gcpv1alpha1 "github.com/crossplaneio/crossplane/pkg/apis/gcp/v1alpha1"
+	"github.com/crossplaneio/crossplane/pkg/clients/gcp/cloudsql"
+	"github.com/crossplaneio/crossplane/pkg/logging"
+	"github.com/crossplaneio/crossplane/pkg/meta"
+	"github.com/crossplaneio/crossplane/pkg/util/googleapi"
+)
+
+const (
+	controllerName = "instance.cloudsql.gcp.crossplane.io"
+	finalizer      = "finalizer." + controllerName
+
+	reconcileTimeout      = 1 * time.Minute
+	requeueAfterOnWait    = 10 * time.Second
+	requeueAfterOnSuccess = 60 * time.Second
+)
+
+var (
+	// requeueNever - object doesn't need any further reconciliations
+	// typically used in terminal cases like: deletion and garbage collection
+	requeueNever = reconcile.Result{}
+
+	// requeueNow - put the object on the work queue immediately
+	// NOTE: this result triggers exponential back-off delay
+	requeueNow = reconcile.Result{Requeue: true}
+
+	// requeueSync - object process was complete and successful and we want to re-sync
+	// remote instance with this object after some delay interval
+	requeueSync = reconcile.Result{RequeueAfter: requeueAfterOnSuccess}
+
+	// requeueWait - object processing was partial, due the remote instance is not being ready, i.e.
+	// performing managedOperations like: create  or update. We want to repeat the processing fo this
+	// object after some short(er) (in comparison to above requeueSync) delay
+	requeueWait = reconcile.Result{RequeueAfter: requeueAfterOnWait}
```

Comment:
I don't think strongly one way or another,  `requeueWait` makes sense to me, but I am open to suggestions.

Reply:
I (as is probably obvious) have a preference for using comments, partly because `requeueAfterWait` vs `requeueAfterSuccess` throws me off because in my head they're both "waiting" to requeue a reconcile, just for different reasons.